framework/wfa/ucc-dummy.py

```diff
@@ -0,0 +1,63 @@
+#!/usr/bin/env python3
+
+import socket
+import argparse
+import sys
+import threading
+import binascii
+
+def handle(host, port, respDict):
+    id = threading.get_ident()
+    print("{}: Listening on {}:{}".format(id, host, port))
+    with socket.socket(socket.AF_INET, socket.SOCK_STREAM) as s:
+        s.bind((host, port))
+        s.listen()
+        conn, addr = s.accept()
+        with conn:
+            print("{}: {}@{} Connected by {}".format(id, host, port, addr))
+            while True:
+                try:
+                    data = conn.recv(1024)
+                except Exception as e:
+                    print("Exception {}".format(e))
+                    break
+                print("{}@{} Received: {}".format(host, port, data))
+                resp = [b'Status,RUNNING', b'Status,COMPLETE'] # default response
+                for key in respDict:
+                    if key in str(data):
+                        resp = respDict[key] # specific resonse
+                for r in resp:
+                    print("{}@{} Sending: {}".format(host, port, r))
+                    conn.sendall(r)
+    print("{}: {}@{} Exiting!".format(id, host, port))
+
+parser = argparse.ArgumentParser()
+parser.add_argument("--ports", nargs='+', default=[7000, 7001, 7002, 7003, 7004, 7005, 7006, 7007, 7008, 7009, 7010, 7011, 7012, 8888])
+parser.add_argument("--host", default='127.0.0.1')
+parser.add_argument("--dst-alid", default="11:22:33:44:55:66")
+args = parser.parse_args()
+
+host = args.host
+ports = [int(p) for p in args.ports]
+alid = args.dst_alid
+
+respDict = {
+    'dev_reset_default' : [b'Status,RUNNING', b'Status,COMPLETE'],
+    'ca_get_version' : [b'Status,RUNNING', b'Status,COMPLETE,vendor,dummy,model,dummy,version,1.0'],
+    'dev_get_parameter' : [b'Status,RUNNING', 'Status,COMPLETE,aLid,{}'.format(alid).encode()],
+}
+
+print("Starting threads on host {}, ports {}".format(host, ports))
+threads = list()
+
+for port in args.ports:
+    print("Creating thread {}@{}".format(host, port))
+    t = threading.Thread(target=handle, args=(host, port, respDict,))

```

Comment:
Personally I'd have used an asyncio server instead of threads, but OK.

Reply:
Will it solve the fact that ctrl+c doesn't work?

Reply:
I guess it would. But I haven't actually used asyncio myself. At least not directly, I've used it through twisted but that hides all the gory details of coroutines and continuations. But basically, with asyncio you still have a single thread and you don't loose your tty association.